services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/EntityManagerImplAutowireTest.java

```diff
@@ -578,76 +604,1068 @@ public void testUpdateEntityJsonWithBooleanList() {
+		// add a list of string boolean annotation
+		toUpdate.put("key", Arrays.asList("true", "false"));
+		// put entity JSON
+		entityManager.updateEntityJson(adminUserInfo, pid, toUpdate);
+		// get the entity JSON
+		JSONObject projectJSON = entityManager.getEntityJson(adminUserInfo, pid, false);
+		assertEquals(projectJSON.getJSONArray("key").get(0).getClass(), String.class);
+		assertEquals(projectJSON.getJSONArray("key").get(1).getClass(), String.class);
+	}
+
+	@Test
+	public void testGetAndUpdateEntityJson_NaN_Infinity() {
+		Project project = new Project();
+		project.setName("test project");
+		String pid = entityManager.createEntity(userInfo, project, null);
+		toDelete.add(pid);
+		project = entityManager.getEntity(userInfo, pid, Project.class);
+		Annotations annotations = entityManager.getAnnotations(userInfo, pid);
+		AnnotationsV2TestUtils.putAnnotations(annotations, "listOfDoubles", Arrays.asList("NaN", "nan", "Infinity", "infinity", "-Infinity", "-infinity"),
+				AnnotationsValueType.DOUBLE);
+		entityManager.updateAnnotations(userInfo, pid, annotations);
+		project = entityManager.getEntity(userInfo, pid, Project.class);
+
+		// Call under test - PLFM-6872
+		// Verify that we get an object back and can read it
+		JSONObject toUpdate = entityManager.getEntityJson(pid, false);
+
+		assertEquals("NaN", toUpdate.getJSONArray("listOfDoubles").get(0));
+		assertEquals("NaN", toUpdate.getJSONArray("listOfDoubles").get(1));
+		assertEquals("Infinity", toUpdate.getJSONArray("listOfDoubles").get(2));
+		assertEquals("Infinity", toUpdate.getJSONArray("listOfDoubles").get(3));
+		assertEquals("-Infinity", toUpdate.getJSONArray("listOfDoubles").get(4));
+		assertEquals("-Infinity", toUpdate.getJSONArray("listOfDoubles").get(5));
+
+
+
+		// Call under test
+		// Verify that we can submit the string representations of the values and they are treated as String,
+		JSONObject result = entityManager.updateEntityJson(userInfo, pid, toUpdate);
+
+		assertNotNull(result);
+
+		JSONArray doubleArray = result.getJSONArray("listOfDoubles");
+		assertNotNull(doubleArray);
+		assertEquals("NaN", doubleArray.get(0));
+		assertEquals("NaN", doubleArray.get(1));
+		assertEquals("Infinity", doubleArray.get(2));
+		assertEquals("Infinity", doubleArray.get(3));
+		assertEquals("-Infinity", doubleArray.get(4));
+		assertEquals("-Infinity", doubleArray.get(5));
+
+		Annotations afterAnnotations = entityManager.getAnnotations(adminUserInfo, pid);
+		assertNotNull(afterAnnotations);
+		assertEquals(project.getId(), afterAnnotations.getId());
+		Map<String, AnnotationsValue> map = afterAnnotations.getAnnotations();
+		assertNotNull(map);
+		assertEquals(1, map.size());
+		AnnotationsValue value = map.get("listOfDoubles");
+		assertEquals(AnnotationsValueType.STRING, value.getType());
+		assertEquals(Arrays.asList("NaN", "NaN", "Infinity", "Infinity", "-Infinity", "-Infinity"), value.getValue());
+	}
+

```

Comment:
test case:
We have a schema but the annotation key is not in the schema.  For that case, it should be treated as if schema did not exist.